### ISSUE TITLE
instance_norm makes a channels_last_3d input contiguous, so a channels-last 3D network pays two layout copies per normalization

### ISSUE THREAD

**Issue by vboussot** [pytorch/pytorch#198375]
### 🐛 Describe the bug

`torch.nn.functional.instance_norm` folds the batch into the channels and runs `batch_norm` on the result. The fold starts with `.contiguous()`:

```cpp
// aten/src/ATen/native/Normalization.cpp:759 (main, 2026-09-23)
auto input_reshaped = input.contiguous().view_symint(shape);
```

For an input in `channels_last_3d` memory format this materializes an NCDHW copy, normalizes it, and returns an NCDHW tensor. The convolution that follows runs in NDHWC when its weight is channels last, so it converts the activation back. A 3D U-Net normalizes after every convolution, which means a `channels_last_3d` network converts each activation twice per block, and the layout that makes its convolutions faster makes the whole forward slower.

This is the shape of every nnU-Net style segmentation model (the ones KonfAI ships): 3D convolutions followed by `InstanceNorm3d`, float16, on patches around 96x128x160.

Repro, eight `Conv3d + InstanceNorm3d + LeakyReLU` blocks of 32 channels on a `[2, 32, 96, 128, 160]` float16 input:

```python
import time, torch

def block(c):
    return torch.nn.Sequential(torch.nn.Conv3d(c, c, 3, padding=1),
                               torch.nn.InstanceNorm3d(c, affine=True), torch.nn.LeakyReLU())

net = torch.nn.Sequential(*[block(32) for _ in range(8)]).cuda().half()
x = torch.randn(2, 32, 96, 128, 160, device="cuda", dtype=torch.half)
for fmt in (torch.contiguous_format, torch.channels_last_3d):
    net.to(memory_format=fmt)
    xf = x.to(memory_format=fmt)
    with torch.inference_mode():
        for _ in range(3):
            net(xf)
        torch.cuda.synchronize()
        t = time.perf_counter()
        for _ in range(10):
            net(xf)
        torch.cuda.synchronize()
        print(fmt, f"{(time.perf_counter() - t) / 10 * 1e3:.1f} ms per forward")

norm = torch.nn.InstanceNorm3d(32, affine=True).cuda().half()
y = norm(x.to(memory_format=torch.channels_last_3d))
print("output channels_last_3d:", y.is_contiguous(memory_format=torch.channels_last_3d))
```

Output on an RTX PRO 5000 (Blackwell, laptop), PyTorch 2.12.1+cu130:

```
torch.contiguous_format 99.5 ms per forward
torch.channels_last_3d 157.1 ms per forward
output channels_last_3d: False
```

The profile of one channels-last forward (`torch.profiler`, CUDA time 142.4 ms) puts 69.9 ms, 49%, in `aten::copy_` under `aten::contiguous`: 15 calls, the 8 inside `instance_norm` and the 7 conversions back for the convolutions that follow. `InstanceNorm3d` alone on that input takes 3.9 ms contiguous and 11.9 ms channels last, and its output is contiguous either way.

Expected: `instance_norm` preserves the input's memory format, like `batch_norm` does (the CUDA batch norm kernels accept channels last). The fold of N into C is a view only for a contiguous NCDHW tensor, but a channels-last input can be normalized per (n, c) without the fold, or handed to `batch_norm` as `(1, N*C, D, H, W)` through a permute that stays a view when the batch is 1.

### Versions

```
PyTorch version: 2.12.1+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 26.04 LTS (x86_64)
GCC version: (Ubuntu 15.2.0-16ubuntu1) 15.2.0
Clang version: Could not collect
CMake version: version 4.2.3
Libc version: glibc-2.43

Python version: 3.13.14 | packaged by conda-forge | (main, Jun 12 2026, 09:50:25) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-7.0.0-31-generic-x86_64-with-glibc2.43
Is CUDA available: True
CUDA runtime version: 13.3.33
GPU models and configuration: GPU 0: NVIDIA RTX PRO 5000 Blackwell Generation Laptop GPU
Nvidia driver version: 595.84
cuDNN version: Could not collect (torch.backends.cudnn.version() reports 92000)
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A
```

cc @jamesr66a
